### PR TITLE
[cli] add documentation for `dns resolve4` command

### PR DESCRIPTION
**Pull request by abtink**

### CODE REVIEW

File: src/cli/README.md [hunk 1]
```diff
@@ -1169,6 +1169,27 @@ DNS response for example.com. - fd4c:9574:3720:2:0:0:5db8:d822 TTL:20456
...
+The parameters after `hostname` are optional. Any unspecified (or zero) value for these optional parameters is replaced by the value from the current default config (`dns config`).
+
+This command requires a NAT64 prefix to be configured and present in Thread Network Data.
+
+The outputted addresses are the NAT64 IPv6 translated versions of the IPv4 addresses from the query response.
+
+```bash
+> route
+2001:db8:122:344::/64 sn med d800
+Done
```
Comment (jwhui) [inline src/cli/README.md:1187]: Is it necessary to explicitly configure this? Shouldn't OTBR configure this automatically?

File: src/cli/README.md [hunk 2]
```diff
@@ -1169,6 +1169,27 @@ DNS response for example.com. - fd4c:9574:3720:2:0:0:5db8:d822 TTL:20456
...
+
+> dns resolve4 ipv6.google.com
+> DNS response for ipv6.google.com - 2001:db8:122:344:c0:2:2100:: TTL: 300
```
Comment (jwhui) [inline src/cli/README.md:1190]: Should we resolve `ipv4.google.com`?

Should we show a /96 prefix? And use an actual IPv4 address that `ipv4.google.com` resolves to today (i.e. 142.250.191.78)?

```suggestion
> dns resolve4 ipv4.google.com
> DNS response for ipv4.google.com - 2001:db8:122:344:c0:2:2100:: TTL: 300
```

File: src/cli/README.md (hunk 1 above)
Comment (abtink) [reply src/cli/README.md]: My idea was to show that we have a NAT64 prefix already (but used the `route` command which shows local netdata). 

I removed it in new push. Added text to mention the NAT64 prefix.

File: src/cli/README.md (hunk 2 above)
Comment (abtink) [reply src/cli/README.md]: Good idea. Changed in new push.

File: src/cli/README.md [hunk 3]
```diff
@@ -1169,6 +1169,25 @@ DNS response for example.com. - fd4c:9574:3720:2:0:0:5db8:d822 TTL:20456
 Done
 ```
 
+### dns resolve4 \<hostname\> \[DNS server IP\] \[DNS server port\] \[response timeout (ms)\] \[max tx attempts\] \[recursion desired (boolean)\]
+
+Send DNS query to obtain IPv4 address for a given hostname.
```
Comment (Vyrastas) [inline src/cli/README.md:1174]: The output doesn't give us the IPv4 address though, so should we revise this to be more in line with what it actually does? (basically the statement on line 1182)

File: src/cli/README.md [hunk 4]
```diff
@@ -1169,6 +1169,25 @@ DNS response for example.com. - fd4c:9574:3720:2:0:0:5db8:d822 TTL:20456
...
+
+Requires `OPENTHREAD_CONFIG_DNS_CLIENT_NAT64_ENABLE`.
+
+The parameters after `hostname` are optional. Any unspecified (or zero) value for these optional parameters is replaced by the value from the current default config (`dns config`).
+
+This command requires a NAT64 prefix to be configured and present in Thread Network Data.
+
+The output provides the NAT64 synthesized IPv6 addresses for the IPv4 addresses from the query response.
+
+Example below is with NAT64 prefix `2001:db8:122:344::/96` being used within Thread mesh. The outputted IPv6 address corresponds to `142.250.191.78` IPv4 address.
```
Comment (Vyrastas) [inline src/cli/README.md:1184]: For example, if a NAT64 prefix of `2001:db8:122:344::/96` is used within the Thread mesh, the outputted IPv6 address corresponds to an IPv4 address of `142.250.191.78` for the `ipv4.google.com` host:

**Review by Vyrastas — CHANGES_REQUESTED** [review]

File: src/cli/README.md (hunk 3 above)
Comment (abtink) [reply src/cli/README.md]: Moved the sentence up.

File: src/cli/README.md (hunk 4 above)
Comment (abtink) [reply src/cli/README.md]: Changed in new push.

**Review by Vyrastas — APPROVED** [review]

**Review by jwhui — APPROVED** [review]

**Pull request merged**